.github/workflows/lock_dev_requirements.yml

```diff
@@ -0,0 +1,40 @@
+---
+name: Lock Dev Requirements
+on:
+  create:
+    branches:
+      - release_**
```

Comment:
Since this only runs on creation of these branches I think this is fine?

The commit seems to land in the right branch in https://github.com/john-westcott-iv/github-file-test, so I think it'll work as intended. @shanemcd are you good with all this?